src/js/AMPSetup/AMPSetup.js

```diff
@@ -78,3 +81,65 @@ AMPHexConsumer.prototype.consumeNibbles = function(nibbles)
 	return this.consumedHex;
 }
 
+/**
+ * This method allows functions to use model constructors even if they were not included in the
+ * correct order in <script> tags.
+ *
+ * @param hexadecimal {AMPHexConsumer} The AMP hex consumer used for constructing the next value from the hexadecimal
+ * @param typeEnumberation {number} The enumeration which represents the data type as defined in the AMP spec.
+ *
+ * @return The constructed model based on the specified data type enumeration. Will have returnValue and nibblesConsumed properties.
+ */
+
+AMPParse.buildUndeclaredType = function(hexadecimal, typeEnumeration)
+{
+	if ( !(hexadecimal instanceof AMPHexConsumer) )
+	{
+		throw new TypeError("Did not receive an instance of AMPHexConsumer");
+	}
+	if (isNaN(typeEnumeration))
+	{
+		throw new TypeError("typeEnumeration is not a number");
+	}
+
```

Comment:
That gets handled in the try/catch below. If we try to access a value not in the AMPBuilders object it'll throw a reference error. It changes the error message when it gets thrown out of this function.